be/src/column/column.h

```diff
@@ -116,6 +116,17 @@ class Column {
     // Current, only support upgrade BinaryColumn to LargeBinaryColumn
     virtual StatusOr<ColumnPtr> upgrade_if_overflow() = 0;
 
+    // Downgrade the column from large column to normal column.
+    // Return internal error if downgrade failed.
+    // Return null, if the column is already normal column, no need to downgrade.
+    // Return the new normal column, if downgrade success
+    // Current, only support downgrade LargeBinaryColumn to BinaryColumn
+    virtual StatusOr<ColumnPtr> downgrade() = 0;
```

Comment:
I have to ask the same question again: since we have both upgrade and downgrade interface, can we hide the implementation inside the column?

Reply:
If this is done, all `Column` operations cannot guarantee that the type is the same, and the type of the Column must be judged.

Reply:
And when the entire SR supports the LargeBinaryColumn, this downgrade can be removed

Reply:
It has been discussed with Mo Fei， A better strategy is needed to solve this problem, but currently we have not find  better way to solve it;